js/src/widget_cell_type.ts

```diff
@@ -0,0 +1,57 @@
+// @ts-ignore
+import * as Handsontable from 'handsontable';
+
+
+async function get_views(model) {
+    let views = [];
+    let id_views = Object.keys(model.views);
+    for (let id of id_views) {
+        views.push(model.views[id]);
+    }
+    return Promise.all(views);
+}
+
+async function widgetRenderer(instance, td, row, col, prop, value, cellProperties) {
+    // If it's a ghost-table, we do not render as it would create a view for nothing
+    // See https://github.com/handsontable/handsontable/blob/6.2.2/src/utils/ghostTable.js#L313-L314
+    // The renderer function is not handled as a Promise, but creating a widget view must be done
+    // with the `create_view` Promise
+    if (td.hasAttribute('ghost-table')) {
+        // TODO Create a div and apply the widget layout to it, also apply the top CSS
+        // class of the widget. How to retrieve CSS classes without a view?
+        return;
+    }
+
+    // Watchdog
+    if (cellProperties['creating_view']) {
+        return;
+    }
+    cellProperties['creating_view'] = true
+
+    // Check if a widget view has already been created, in this case do nothing
+    let views = await get_views(value);
+    for (let child of td.children) {
+        for (let view of views) {
+            // If it's a view of this widget, do nothing
+            if (view.el === child) {
+                cellProperties['creating_view'] = false
+                return;
+            }
+        }
+        // If it's not a view of this widget, we remove it
+        td.removeChild(child);
+    }
+
+    return value.widget_manager.create_view(value).then((view) => {
+        td.appendChild(view.el);
+        cellProperties['creating_view'] = false
+        // The following is needed until we find a way to fix the issue with
+        // ghost-table
+        view.el.style.width = '100%';
+        view.el.style.height = '100%';
+    });
```

Comment:
I am not so sure about this render, I'm tweaking and testing it now, will get back to it later.